### PR TITLE
LPS-68543 - Session isn't auto extended when session.timeout.auto.extend is set to true

### PR DESCRIPTION
**Pull request by antonio-ortega**
Hi again Jon,

Following @jbalsas' advice, I've rebuilt the PR removing some old and unneeded commits to make it easier to understand.  I'll close [the previous PR](https://github.com/jonmak08/liferay-portal/pull/1658) I sent to you this week.

Regards,

### CODE REVIEW

**Comment by antonio-ortega** [comment]
ci:reopen

**Comment by antonio-ortega** [comment]
Hi again,

I'm reopening the PR because the SF error is not related to this PR. Please, review it. Thanks in advance!

**Comment by jonmak08** [comment]
Hey @antonio-ortega,

Sorry for the delay,  we had a MetalJS training last week and I am finishing up year end reviews this week.  I should be able to get to this review by tomorrow for you.

Please let me know if you have any questions. Thanks!

File: modules/apps/foundation/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/session.js [hunk 1]
```diff
@@ -328,46 +328,42 @@ AUI.add(
...
-										else if (timeOffset < warningTime) {
-											hasWarned = false;
-											updateSessionState = false;
-										}
+									if (timestamp == 'expired') {
+										expirationMoment = true;
+										hasExpired = true;
 									}
 
 									if (updateSessionState) {
```
Comment (jonmak08) [inline modules/apps/foundation/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/session.js]: This is always set to ```true``` (See https://github.com/jonmak08/liferay-portal/pull/1662/commits/96f12419597d08640ee2c4bc109b58d79be730af#diff-1ad2f307ca967b71dac6d9fa38581166L339).  What is the purpose in having this line  now that is will always be set to ```true```?

File: modules/apps/foundation/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/session.js [hunk 2]
```diff
@@ -328,46 +328,42 @@ AUI.add(
...
 										var sessionState = instance.get('sessionState');
 
 										if (hasExpired && sessionState != 'expired') {
-											instance.expire();
+											if (extend) {
+
+												hasExpired = false;
+												hasWarned = false;
+												expirationMoment = false;
+												warningMoment = false;
```
Comment (jonmak08) [inline modules/apps/foundation/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/session.js]: Alphabetize these JS variables.

File: modules/apps/foundation/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/session.js [hunk 3]
```diff
@@ -328,46 +328,42 @@ AUI.add(
...
+										hasExpired = true;
 									}
 
 									if (updateSessionState) {
 										var sessionState = instance.get('sessionState');
 
 										if (hasExpired && sessionState != 'expired') {
-											instance.expire();
+											if (extend) {
+
```
Comment (jonmak08) [inline modules/apps/foundation/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/session.js]: Remove extra line break.

File: modules/apps/foundation/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/session.js [hunk 4]
```diff
@@ -328,46 +328,42 @@ AUI.add(
...
+												hasExpired = false;
+												hasWarned = false;
+												expirationMoment = false;
+												warningMoment = false;
+
+												instance.extend();
+											} else {
+												instance.expire();
```
Comment (jonmak08) [inline modules/apps/foundation/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/session.js]: Add line break between different JS statements.

**Pull request closed without merging**